packages/cli-kit/src/public/node/http.test.ts

```diff
@@ -1,28 +1,231 @@
+import {downloadFile, shopifyFetch, formData, requestMode, fetch} from './http.js'
+import {mockAndCaptureOutput} from './testing/output.js'
 import {fileExists, inTemporaryDirectory, readFile} from './fs.js'
-import {downloadFile} from './http.js'
 import {joinPath} from './path.js'
-import {describe, expect, test, vi} from 'vitest'
-import nodeFetch, {Response} from 'node-fetch'
-import {Readable} from 'stream'
+import {getAllPublicMetadata} from './metadata.js'
+import {skipNetworkLevelRetry} from './environment.js'
+import {platformAndArch} from './os.js'
+import {afterAll, afterEach, beforeAll, beforeEach, describe, expect, test, vi} from 'vitest'
+import {setupServer} from 'msw/node'
+import {delay, http, HttpResponse} from 'msw'
+import FormData from 'form-data'
 
-vi.mock('node-fetch', async () => {
-  const actual: any = await vi.importActual('node-fetch')
+const DURATION_UNTIL_ABORT_IS_SEEN = 100
+
+vi.mock('./environment.js', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('./environment.js')>()
+
+  const skipNetworkLevelRetryMock = vi.fn().mockReturnValue(false)
```

Comment:
instead of mocking this, we can inject `process.env` to the `requestMode` function. 

```ts
export function requestMode(preset: RequestModeInput = 'default', processEnv = process.env): RequestBehaviour {
```